### PR TITLE
add seq file load for image classification

### PR DESCRIPTION
**Pull request by zhangxiaoli73**

### CODE REVIEW

File: dl/src/test/scala/com/intel/analytics/bigdl/dataset/TransformersSpec.scala [hunk 1]
```diff
@@ -431,7 +431,7 @@ class TransformersSpec extends FlatSpec with Matchers {
 
   "RGBImage To SeqFile" should "be good" in {
```
Comment (yiheng) [inline dl/src/test/scala/com/intel/analytics/bigdl/dataset/TransformersSpec.scala]: Please add new unit test for sequence file with name case

File: dl/src/main/scala/com/intel/analytics/bigdl/dataset/image/LocalScaleImgReaderSeq.scala [hunk 2]
```diff
@@ -0,0 +1,77 @@
...
+package com.intel.analytics.bigdl.dataset.image
+
+import java.awt.color.ColorSpace
+
+import com.intel.analytics.bigdl.dataset.Transformer
+
+import scala.collection.Iterator
+
+
+object LocalImgReaderSeq {
```
Comment (yiheng) [inline dl/src/main/scala/com/intel/analytics/bigdl/dataset/image/LocalScaleImgReaderSeq.scala]: The name is confusing.

File: dl/src/main/scala/com/intel/analytics/bigdl/dataset/image/LocalScaleImgReaderSeq.scala [hunk 3]
```diff
@@ -0,0 +1,77 @@
...
+  def apply(scaleTo: Int = BGRImage.NO_SCALE, normalize: Float = 255f)
+  : Transformer[LocalLabeledImagePath, (LabeledBGRImage, String)]
+  = new LocalScaleImgReaderSeq(scaleTo, normalize)
+
+  def apply(resizeW: Int, resizeH: Int, normalize: Float)
+  : Transformer[LocalLabeledImagePath, (LabeledBGRImage, String)]
+  = new LocalResizeImgReaderSeq(resizeW, resizeH, normalize)
+}
+
+/**
```
Comment (yiheng) [inline dl/src/main/scala/com/intel/analytics/bigdl/dataset/image/LocalScaleImgReaderSeq.scala]: You need update the comments.

File: dl/src/test/scala/com/intel/analytics/bigdl/dataset/TransformersSpec.scala (hunk 1 above)
Comment (zhangxiaoli73) [inline dl/src/test/scala/com/intel/analytics/bigdl/dataset/TransformersSpec.scala]: ok. add a unit test named "RGBImage To SeqFile with name"

File: dl/src/main/scala/com/intel/analytics/bigdl/dataset/image/LocalScaleImgReaderSeq.scala (hunk 2 above)
Comment (zhangxiaoli73) [inline dl/src/main/scala/com/intel/analytics/bigdl/dataset/image/LocalScaleImgReaderSeq.scala]: ok, chang to "LocalImgReaderWithName"

File: dl/src/main/scala/com/intel/analytics/bigdl/dataset/image/LocalScaleImgReaderSeq.scala (hunk 3 above)
Comment (zhangxiaoli73) [inline dl/src/main/scala/com/intel/analytics/bigdl/dataset/image/LocalScaleImgReaderSeq.scala]: ok, add some comments

File: dl/src/main/scala/com/intel/analytics/bigdl/dataset/image/LocalImgReaderWithName.scala [hunk 4]
```diff
@@ -0,0 +1,80 @@
...
+
+import java.awt.color.ColorSpace
+
+import com.intel.analytics.bigdl.dataset.Transformer
+
+import scala.collection.Iterator
+
+/**
+ * Read BGR images from given paths,
+ * return LabeledBGRImage and the file name that can add to the sequence file.
```
Comment (zhangxiaoli73) [inline dl/src/main/scala/com/intel/analytics/bigdl/dataset/image/LocalImgReaderWithName.scala]: sorry, changed

File: dl/src/main/scala/com/intel/analytics/bigdl/dataset/image/LocalImgReaderWithName.scala [hunk 5]
```diff
@@ -0,0 +1,80 @@
...
+  def apply(resizeW: Int, resizeH: Int, normalize: Float)
+  : Transformer[LocalLabeledImagePath, (LabeledBGRImage, String)]
+  = new LocalResizeImgWithName(resizeW, resizeH, normalize)
+}
+
+/**
+ * Read BGR images from given paths. After read the image, it will resize the shorted edge to the
+ * given scale to value and resize the other edge properly. It will also divide the pixel value
+ * by the given normalize value.
+ * Besides, it will return the file name that can add to the sequence file.
```
Comment (zhangxiaoli73) [inline dl/src/main/scala/com/intel/analytics/bigdl/dataset/image/LocalImgReaderWithName.scala]: changed

File: dl/src/main/scala/com/intel/analytics/bigdl/dataset/image/BGRImgToLocalSeqFile.scala [hunk 6]
```diff
@@ -56,15 +56,19 @@ class BGRImgToLocalSeqFile(blockSize: Int, baseFileName: Path) extends
           SequenceFile.Writer.valueClass(classOf[Text]))
         var i = 0
         while (i < blockSize && prev.hasNext) {
-          val image = prev.next()
+          val iter = prev.next()
```
Comment (zhangxiaoli73) [inline dl/src/main/scala/com/intel/analytics/bigdl/dataset/image/BGRImgToLocalSeqFile.scala]: ok, changed

File: dl/src/main/scala/com/intel/analytics/bigdl/dataset/DataSet.scala [hunk 7]
```diff
@@ -401,6 +399,20 @@ object DataSet {
       }
     }
 
+    def getValue(data: Text, dataType: String): String = {
```
Comment (zhangxiaoli73) [inline dl/src/main/scala/com/intel/analytics/bigdl/dataset/DataSet.scala]: yeah,  readLabel and readName is more clear for user. OK, changed.

**Comment by yiheng** [comment]
can you merge the 4 commits into one?

**Comment by zhangxiaoli73** [comment]
ok, merge commits into one.

**Pull request merged**